src/main/scala/io/iohk/praos/domain/Block.scala

```diff
@@ -0,0 +1,21 @@
+package io.iohk.praos.domain
+
+import io.iohk.praos.crypto.{Hasher, Seed, Signature, VrfProof, combineSeeds}
+
+
+case class Block(
+  state       : Option[Hasher#Digest],
```

Comment:
Maybe we could define `state: Option[ByteString]`? The type of the hash function probably won't change, so we might avoid having `type Digest` in `Hasher`.

Reply:
What if we want to use, say, `Array[Byte]` or similar in the future for whatever reason? IMHO, even when the result type of the hash function isn't likely to change, having a type `Hasher#Digest` is way more expressive than using a plain `ByteString`.